sdk/python/lib/pulumi/automation/_stack.py

```diff
@@ -606,6 +639,85 @@ def destroy(
             stdout=destroy_result.stdout, stderr=destroy_result.stderr, summary=summary
         )
 
+    def import_resources(
+        self,
+        message: Optional[str] = None,
+        resources: Optional[List[ImportResource]] = None,
+        nameTable: Optional[Dict[str, str]] = None,
+        protect: Optional[bool] = None,
+        generate_code: Optional[bool] = None,
+        converter: Optional[str] = None,
+        converter_args: Optional[List[str]] = None,
+        on_output: Optional[OnOutput] = None,
+        show_secrets: Optional[bool] = True,
```

Comment:
I don't know what Python says, but if we're defaulting to `True`, is this really `Optional[bool]`, or can it be `bool`? I don't know what the rules are. Either way, if we are going to keep this as a default to `True` (and not `None`), I'd be minded to apply the same pattern to `protect` and `generate_code`, rather than introducing `None` there. It should (I think) clean up some of the fiddlier checks you have later on (where you have to be careful not to confuse `None` and `False`).

Reply:
> if we're defaulting to True, is this really Optional[bool], or can it be bool?

Changed to just `bool` because we are defaulting. 

As for `protect` and `generate_code` I would rather keep it optional because it reflects the usage of the CLI, and the CLI decides the default